File: Project/Backend/src/ArtAuction.Infrastructure/Repositories/CategoryRepo.cs
using ArtAuction.Application.Entities;
using ArtAuction.Application.Interfaces.Repositories;
using ArtAuction.Infrastructure.Persistence;
using Microsoft.EntityFrameworkCore;

namespace ArtAuction.Infrastructure.Repositories;

public class CategoryRepo : ICategoryRepo
{
    // Attributes
    private readonly AppDbContext _dbContext;

    // COnstructor
    public CategoryRepo(AppDbContext dbContext)
    {
        this._dbContext = dbContext;
    }

    // Method
    public async Task<ICollection<Category>> GetAllCategories()
    {
        return await _dbContext.Categories.AsNoTracking().ToListAsync();
    }
}

File: ArtworkPostServices/src/main/java/com/artauction/artworkpostservices/Repositories/Interfaces/CategoryRepo.java
package com.artauction.artworkpostservices.Repositories.Interfaces;

import com.artauction.artworkpostservices.Entities.Category;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CategoryRepo extends JpaRepository<Category, Integer> {
}
